config/routes/system_admin_routes.rb

```diff
@@ -4,7 +4,7 @@ module SystemAdminRoutes
   def self.extended(router)
     router.instance_exec do
       scope module: :system_admin, path: "system-admin" do
-        resources :providers, except: %i[edit update destroy] do
+        resources :providers, except: %i[destroy] do
```

Comment:
Can you change this to `only` instead of `except`. Its better to be explicit about which actions we want.